python/lsst/obs/base/gen3/ingest_tests.py

```diff
@@ -0,0 +1,200 @@
+# This file is part of obs_subaru.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Base class for writing Gen3 raw data ingest tests.
+"""
+
+__all__ = ("IngestTestBase",)
+
+import abc
+import tempfile
+import os
+import shutil
+
+from lsst.daf.butler import Butler
+from lsst.obs.base.gen3 import RawIngestTask
+
+
+class IngestTestBase(metaclass=abc.ABCMeta):
+    """Base class for tests of gen3 ingest.
+    """
+
+    ingestDir = ""
+    """Root path to ingest files into. Typically `obs_package/tests/`; the
+    actual directory will be a tempdir under this one.
+    """
+
+    instrument = None
+    """The instrument to be registered and tested."""
+
+    dataId = {}
+    """Butler dataId of a file to ingest when testing."""
```

Comment:
Should add a type here in parens and backticts, especially because daf_butler has some special classes to represent validated/expanded data IDs but this looks like it's a vanilla dict (which is fine).  For consistency I'd recommend just saying "data ID" instead of "dataId" earlier in the sentence too, as that's how we usually spell the concept as opposed to a variable or class representing one.